youtube_dl/extractor/panopto.py

```diff
@@ -0,0 +1,607 @@
+        formats = []
+        subtitles = {}
+        for stream in streams or []:
+            stream_formats = []
+            http_stream_url = stream.get('StreamHttpUrl')
+            stream_url = stream.get('StreamUrl')
+
+            if http_stream_url:
+                stream_formats.append({'url': http_stream_url})
+
+            if stream_url:
+                media_type = stream.get('ViewerMediaFileTypeName')
+                if media_type in ('hls', ):
+                    m3u8_formats, stream_subtitles = self._extract_m3u8_formats_and_subtitles(stream_url, video_id)
+                    stream_formats.extend(m3u8_formats)
+                    subtitles = self._merge_subtitles(subtitles, stream_subtitles)
+                else:
+                    stream_formats.append({
+                        'url': stream_url
+                    })
+            for fmt in stream_formats:
+                fmt.update({
+                    'format_note': stream.get('Tag'),
+                    **fmt_kwargs
+                })
+            formats.extend(stream_formats)
+
+        return formats, subtitles
+
+    def _real_extract(self, url):
+        base_url, video_id = self._match_valid_url(url).group('base_url', 'id')
+        delivery_info = self._call_api(
+            base_url, '/Pages/Viewer/DeliveryInfo.aspx', video_id,
+            query={
+                'deliveryId': video_id,
+                'invocationId': '',
+                'isLiveNotes': 'false',
+                'refreshAuthCookie': 'true',
+                'isActiveBroadcast': 'false',
+                'isEditing': 'false',
+                'isKollectiveAgentInstalled': 'false',
+                'isEmbed': 'false',
+                'responseType': 'json',
+            }
+        )
+
+        delivery = delivery_info['Delivery']
+        session_start_time = int_or_none(delivery.get('SessionStartTime'))
+        timestamps = delivery.get('Timestamps')
+
+        # Podcast stream is usually the combined streams. We will prefer that by default.
+        podcast_formats, podcast_subtitles = self._extract_streams_formats_and_subtitles(
+            video_id, delivery.get('PodcastStreams'), format_note='PODCAST')
+
+        streams_formats, streams_subtitles = self._extract_streams_formats_and_subtitles(
+            video_id, delivery.get('Streams'), preference=-10)
+
+        formats = podcast_formats + streams_formats
+        formats.extend(self._extract_mhtml_formats(base_url, timestamps))
+        subtitles = self._merge_subtitles(
+            podcast_subtitles, streams_subtitles, self.extract_subtitles(base_url, video_id, delivery))
+
+        self._sort_formats(formats)
+        self.mark_watched(base_url, video_id, delivery_info)
+
+        return {
+            'id': video_id,
+            'title': delivery.get('SessionName'),
+            'cast': traverse_obj(delivery, ('Contributors', ..., 'DisplayName'), default=[], expected_type=lambda x: x or None),
```

Comment:
What do you think `expected_type=lambda x: x or None` is doing? Apparently, a callable `expected_type` only fails if it returns None, so this is filtering out any falsy values!